Doctests.hs

```diff
@@ -0,0 +1,12 @@
+{-# OPTIONS_GHC -Wall #-}
```

Comment:
Can this live in a new test directory, and be run automatically when stack test is called?